include/caf/sensor_sampler.h

```diff
@@ -41,6 +41,7 @@ struct sensor_config {
 	uint8_t chan_cnt;
 	unsigned int sampling_period_ms;
 	struct trigger *trigger;
+	int active_events_limit;
```

Comment:
`size_t`? We may also consider using uint16_t/uint8_t and place it after chan_cnt to save some memory :)